geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/Z2Table.scala

```diff
@@ -0,0 +1,190 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.accumulo.data.tables
+
+
+import java.nio.charset.StandardCharsets
+
+import com.google.common.collect.{ImmutableSet, ImmutableSortedSet}
+import com.google.common.primitives.{Bytes, Longs}
+import com.vividsolutions.jts.geom.{Geometry, GeometryCollection, LineString, Point}
+import org.apache.accumulo.core.client.admin.TableOperations
+import org.apache.accumulo.core.conf.Property
+import org.apache.accumulo.core.data.{Mutation, Value}
+import org.apache.hadoop.io.Text
+import org.locationtech.geomesa.accumulo.data.AccumuloFeatureWriter.{FeatureToMutations, FeatureToWrite}
+import org.locationtech.geomesa.accumulo.data.EMPTY_TEXT
+import org.locationtech.geomesa.curve.Z2SFC
+import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+import org.locationtech.sfcurve.zorder.Z2
+import org.locationtech.sfcurve.zorder.Z3.ZPrefix
+import org.opengis.feature.simple.SimpleFeatureType
+
+object Z2Table extends GeoMesaTable {
+
+  val FULL_CF = new Text("F")
+  val BIN_CF = new Text("B")
+  val EMPTY_BYTES = Array.empty[Byte]
+  val EMPTY_VALUE = new Value(EMPTY_BYTES)
+  val NUM_SPLITS = 4 // can't be more than Byte.MaxValue (127)
+  val SPLIT_ARRAYS = (0 until NUM_SPLITS).map(_.toByte).toArray.map(Array(_)).toSeq
+
+  // the bytes of z we keep for complex geoms
+  // 3 bytes is 22 bits of geometry (not including the first 2 bits which aren't used)
+  // roughly equivalent to 4 digits of geohash (32^4 == 2^20) and ~20km resolution
+  val GEOM_Z_NUM_BYTES = 3
+  // mask for zeroing the last (8 - GEOM_Z_NUM_BYTES) bytes
+  val GEOM_Z_MASK: Long =
+    java.lang.Long.decode("0x" + Array.fill(GEOM_Z_NUM_BYTES)("ff").mkString) << (8 - GEOM_Z_NUM_BYTES) * 8
+
+  override def supports(sft: SimpleFeatureType): Boolean =
+    sft.getGeometryDescriptor != null && sft.getSchemaVersion > 7
```

Comment:
Do we have a list of constants somewhere for these versions?  It might be nice to organize that just to make clear what features started/ended with particular version numbers.